app/src/main/kotlin/com/wire/android/feature/conversation/content/domain/MessageSender.kt

```diff
@@ -0,0 +1,114 @@
+package com.wire.android.feature.conversation.content.domain
+
+import com.wire.android.core.exception.Failure
+import com.wire.android.core.exception.NetworkConnection
+import com.wire.android.core.exception.Unauthorized
+import com.wire.android.core.functional.Either
+import com.wire.android.core.functional.suspending
+import com.wire.android.core.network.NetworkHandler
+import com.wire.android.feature.contact.DetailedContact
+import com.wire.android.feature.conversation.content.Message
+import com.wire.android.feature.conversation.content.MessageRepository
+import com.wire.android.feature.conversation.content.SendMessageFailure
+import com.wire.android.feature.messaging.ChatMessageEnvelope
+import com.wire.android.feature.messaging.ClientPayload
+import com.wire.android.feature.messaging.RecipientEntry
+import com.wire.android.shared.session.SessionRepository
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.withContext
+
+class MessageSender(
```

Comment:
This class has 2 responsibilities. It sends message and creates envelope
Why not to extract the envelope creation method to a separate class

Reply:
It could be split...

I have been fantasising about this in my head for a while. The main drawback at the moment of doing something like that is growing even further the list of dependencies. `MessageSender` already has 5 dependencies, and at least one more is needed for the single thread coroutine context.

Maybe `networkHandler` can be removed. As `trySendingOutgoingMessage` is only supposed to be called:
1. When `SendMessageService` calls it (after checking for internet connection).
2. When WorkManager launches `AndroidSendMessageWorker`.

Also, it is expected that a network failure can happen during the sending process. So it doesn't make much sense to check for it inside `MessageSender`.

Reply:
With all of that in mind. My propose is:

1. Remove `networkHandler` from the class and the checking for internet connection, as it is very redundant.
2. Move the `Envelope` creation to another specialised class.

Reply:
Yeah, I think It would be better like this

Reply:
Done! Check #295 for the envelope creator. Let's merge that before this one.